pgdog/src/backend/pool/address.rs

```diff
@@ -48,9 +50,9 @@ impl Address {
         }
     }
 
-    /// Get address for `TCPStream`.
-    pub fn addr(&self) -> String {
-        format!("{}:{}", self.host, self.port)
+    pub async fn addr(&self) -> Result<SocketAddr, std::io::Error> {
+        let ip = DnsCache::global().resolve(&self.host).await?;
```

Comment:
For this one, you might want to do something like this:

1. Check if DNS cache is enabled (setting is not `None`)
2. If `None` (disabled), just return host + port like before, letting `TcpStream` resolve it like before using the OS resolver
3. If enabled, use the `DnsCache`